src/bootman/bootman.c

```diff
@@ -305,6 +305,31 @@ bool boot_manager_remove_kernel(BootManager *self, const Kernel *kernel)
         return self->bootloader->remove_kernel(self, kernel);
 }
 
+int detect_and_mount_boot(BootManager *self, char **boot_dir) {
+        autofree(char) *boot_dev = NULL;
+        const char *prefix;
+        int wanted_boot_mask;
+
+        wanted_boot_mask = self->sysconfig->wanted_boot_mask;
+        if ((wanted_boot_mask & BOOTLOADER_CAP_LEGACY) != BOOTLOADER_CAP_LEGACY) {
+                return mount_boot(self, boot_dir);
+        }
+
+        boot_dev = get_boot_device();
+
+        if (!boot_dev) {
+                prefix = boot_manager_get_prefix((BootManager *)self);
+                boot_dev = get_legacy_boot_device((char *)prefix);
+        }
+
+        if (!boot_dev) {
+                LOG_ERROR("Failed to detect boot device");
+                return -1;
+        }
+
+        return mount_boot(self, boot_dir);
```

Comment:
In the case where there isn't a boot partition isn't this going to try and mount the root partition?

Reply:
yep, you're right, I should return 0 instead of -1 on line 327. Fixing. Thanks.

Reply:
Looking at mount_boot, it looks like the eventual call to cbm_get_mountpoint_for_device would check to see if the sysconfig->boot_device is already mounted and will just reuse it but that would point to the rootfs and the code in cbm_get_mountpoint_for_device is really assuming an ESP.

Reply:
How is it assuming ESP? I'm looking into `cbm_get_mountpoint_for_device()` and can't see any indication of that.